CustomBatteries/API/CbItem.cs

```diff
@@ -0,0 +1,113 @@
+﻿namespace CustomBatteries.API
+{
+    using System.Collections.Generic;
+    using UnityEngine;
+    using Logger = QModManager.Utility.Logger;
+#if SUBNAUTICA
+    using Sprite = Atlas.Sprite;
+#endif
+
+    /// <summary>
+    /// An interface that defines all the necessary elements of a CustomBatteries plugin pack.
+    /// </summary>
+    public abstract class CbItem
+    {
+        /// <summary>
+        /// The full capacity of energy of the item.
+        /// </summary>
+        public abstract int EnergyCapacity { get; }
+
+        /// <summary>
+        /// The internal ID for the custom item.
+        /// </summary>
+        public abstract string ID { get; }
+
+        /// <summary>
+        /// The display name of the custom item shown in-game.
+        /// </summary>
+        public abstract string Name { get; }
+
+        /// <summary>
+        /// The flavor text for the custom item shown in-game when viewing it from a PDA screen.
+        /// </summary>
+        public abstract string FlavorText { get; }
+
+        /// <summary>
+        /// The materials required to craft the item.<para/>
+        /// If you want multiple copies of the same material, include multiple entries of that <see cref="TechType"/>.<para/>
+        /// If this list is empty, a default recipe of a single <see cref="TechType.Titanium"/> will be applied instead.
+        /// </summary>
+        public abstract IList<TechType> CraftingMaterials { get; }
+
+        /// <summary>
+        /// What item must be obtained, scanned, or built to unlock the battery and power cell.<para/>
+        /// This property is optional. By default, the item will be unlocked at the start of the game.
+        /// </summary>
+        public virtual TechType UnlocksWith => TechType.None;
+
+        /// <summary>
+        /// The custom sprite for the item's icon.<br/>
+        /// This property is optional and will default to the standard icon for batteries or power cells.
+        /// </summary>
+        public virtual Sprite CustomIcon => null;
+
+
+        /// <summary>
+        /// The custom data that will make up your batteries model.<br/>
+        /// This property is optional and will default to the standard model for batteries or power cells if left as null.
+        /// </summary>
+        public virtual CBModelData CBModelData => null;
+
+        /// <summary>
+        /// Override this value if you want your item to not be allowed in Battery and Power Cell chargers.
+        /// </summary>
+        public virtual bool ExcludeFromChargers => false;
```

Comment:
This feature should be removed until such a time as it actually provides more solutions than problems.

Reply:
if its just going to cause problems then yea. and people that want special conditions can just patch the code themselves to remove them.